package/MDAnalysis/analysis/base.py

```diff
@@ -85,21 +84,198 @@ def _prepare(self):
 
     def _conclude(self):
         """Finalise the results you've gathered.
-
         Called at the end of the run() method to finish everything up.
         """
         pass
 
-    def run(self):
+    def run(self, parallel=None, threads=None):
+        """ Chooses whether to run analysis in serial or parallel
+        mode depending on user input"""
+        if parallel is None:
+            self._serial_run()
+        else:
+            self._parallel_run(threads)
+
+    def _serial_run(self):
         """Perform the calculation"""
         logger.info("Starting preparation")
+        prog = mda.lib.log.Progressbar([self.nframes])
+        prog.start()
         self._prepare()
         for i, ts in enumerate(
                 self._trajectory[self.start:self.stop:self.step]):
-            self._ts = ts
             #logger.info("--> Doing frame {} of {}".format(i+1, self.nframes))
-            self._single_frame()
-        logger.info("Finishing up")
+            self._single_frame(ts)
+            prog.update(0)
+        #logger.info("Finishing up")
+        self._conclude()
+
+    def _parallel_run(self, threads=None):
+        """
+        Create copies of the original object to be
+        dispatched to multiprocess
+        """
+
+        if threads is None:
+            self.threads = mp.cpu_count()-1
```

Comment:
Not stupid at all! When testing i found out that using all cores lead to worst performance than all-1. I think this is because the "main" process has its own thread as well, thus you end up using ncores+1, which slows down everything. I tested this on a 48 core machine so maybe it was a bit extreme, I'll try on a machine where the cpu count is smaller.